rs/execution_environment/src/scheduler.rs

```diff
@@ -2461,6 +2461,17 @@ fn is_next_method_chosen(
     may_schedule_heartbeat: bool,
     may_schedule_global_timer: bool,
 ) -> bool {
+    // If the task on the front of the task queue is hook,
```

Comment:
Maybe you adjust the doc comment of the method instead of adding this here so it's all contained in one place?